src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs

```diff
@@ -429,6 +434,11 @@ internal sealed class W3dModelDrawConditionState : DisposableBase
 
         public bool StillActive() => Model.AnimationInstances.Any(x => x.IsPlaying);
 
+        public new void Dispose()
+        {
+            Deactivate();
+        }
```

Comment:
Perhaps instead of overwriting the `Dispose` method, we could use `DisposableBase.AddDisposeAction`?

Better yet, `AddDisposeAction` could happen within the `ParticleSystem` itself, and then the particle systems here could just be added via `DisposableBase.AddDisposable`?